Generate scopes

karczex opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/pmemkv-bench/68)
<!-- Reviewable:end -->

Commit:
Extend numactl handling in run_benchmark.py

Allow to set all numactl parameters in pmemkv-bench configuration file

igchor reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-:-MWdCbPwARaAyfsYgg5Q:bnfp4nl)* status: 5 of 6 files reviewed, all discussions resolved (waiting on @karczex)



<!-- Sent from Reviewable.io -->

lukaszstolarczuk reviewed (commented):
besides minor issues, I like it a lot!
#goodchange ;)

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-:-MWdpLvNE5PXIsw1t2SF:brz0h8x)* status: all files reviewed, 11 unresolved discussions (waiting on @karczex)

___
*[run_benchmark.py, line 27 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdn1RH1q3t9Tcfvs50:-MWdn1RH1q3t9Tcfvs51:bikucen) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/run_benchmark.py#L27)):*
> ```Python
> 
> 
> class CmdLine:
> ```

maybe just a short description, it's not obvious what's this for (I guess)
___
*[run_benchmark.py, line 45 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdn0Fl6JEyaI_uhmim:-MWdn0Fl6JEyaI_uhmin:b-mchm1z) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/run_benchmark.py#L45)):*
> ```Python
>         return " ".join(self)
> 
>     def __getitem__(self, item):
> ```

do we need this?
___
*[run_benchmark.py, line 161 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdkVES3BivFmDTZFoM:-MWdkVES3BivFmDTZFoN:bb5zb8q) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/run_benchmark.py#L161)):*
> ```Python
>             raise e
> 
>     def run(self, environ, benchmark_params, numactl=None):
> ```

perhaps rename to `numactl_params`
___
*[bench_scenarios/generate_dram_scope.py, line 12 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdjT9V96dJQdWjFy8H:-MWdjT9V96dJQdWjFy8I:b-sh6r71) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L12)):*
> ```Python
> import json
> import itertools
> import jsonschema
> ```

\- jsonschema
___
*[bench_scenarios/generate_dram_scope.py, line 28 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnD564yG72dsqkUj2:-MWdnD564yG72dsqkUj3:bwle16t) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L28)):*
> ```Python
> key_size = [8]
> value_size = [8, 128, 1024]
> number_of_elements = 1000000
> ```

only one Mill. ? perhaps more...?
___
*[bench_scenarios/generate_dram_scope.py, line 51 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnQgQ8F8Px9ULwo9q:-MWdnQgQ8F8Px9ULwo9r:b65za00) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L51)):*
> ```Python
>             "pmemkv_bench": {
>                 "--benchmarks": f"{benchmark[0]}",
>                 "--value_size": f"{benchmark[1]}",
> ```

missing key_size
___
*[bench_scenarios/generate_memkind_based_scope.py, line 30 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnXyk6jnSmxDVE6lv:-MWdnXyk6jnSmxDVE6lw:b4uqd4d) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_memkind_based_scope.py#L30)):*
> ```Python
> 
> def concurrent_engines():
> ```

empty line, pls remove
___
*[bench_scenarios/generate_memkind_based_scope.py, line 67 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdno2r0UM4KrgjQ2nL:-MWdno2r0UM4KrgjQ2nM:bcghe2z) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_memkind_based_scope.py#L67)):*
> ```Python
>                 "--num": f"{number_of_elements}",
>                 "--db": db_path,
>                 "--db_size_in_gb": "500",
> ```

make it a const on top, pls
___
*[bench_scenarios/generate_memkind_based_scope.py, line 70 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnslF1cbUsn8Rf97g:-MWdnslF1cbUsn8Rf97h:b-lq0ip1) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_memkind_based_scope.py#L70)):*
> ```Python
>             },
>             "numactl": {
>                 "--cpubind": f"file:{os.path.dirname(db_path)}",
> ```

just want to double check this - have you tested it works on the machine, you'll be running the benchmark...? (so e.g. the weekend execution wouldn't be to waste...)
___
*[bench_scenarios/generate_obj_based_scope.py, line 76 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdo8lf0bx4YAcEPF1Q:-MWdo8lf0bx4YAcEPF1R:b3kqx) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_obj_based_scope.py#L76)):*
> ```Python
>                 "--num": f"{number_of_elements}",
>                 "--db": db_path,
>                 "--db_size_in_gb": "500",
> ```

.
___
*[tests/test.py, line 14 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdp39s73uEohddaT9O:-MWdp39s73uEohddaT9P:b-6bga4j) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L14)):*
> ```Python
> project_path = os.path.dirname(tests_path)
> sys.path.append(project_path)
> import run_benchmark as rb
> ```

move imports to top...?
___
*[tests/test.py, line 34 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdosIIBdKt8HLk6Xxx:-MWdosIIBdKt8HLk6Xxy:br2ppgr) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L34)):*
> ```Python
>             "-DENGINE_ROBINHOOD=1",
>             "-DENGINE_VCMAP=1",
>             "-DBUILD_JSON_CONFIG=1",
> ```

perhaps add also DRAM engine...?
___
*[tests/test.py, line 68 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdopBw-q3ELsDQ1z-B:-MWdopBw-q3ELsDQ1z-C:bn257cr) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L68)):*
> ```Python
>     test_config_file = create_config_file(benchmark_configuration)
>     sys.argv = ["dummy.py", build_config_file.name, test_config_file.name]
>     try:
> ```

maybe add here schema validation, as well?


<!-- Sent from Reviewable.io -->

lukaszstolarczuk reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: all files reviewed, 12 unresolved discussions (waiting on @karczex)

*[a discussion](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdq49L2jbYl-aQK219:-MWdq49L2jbYl-aQK21A:b-l7d5up) (no related file):*
hmm and for some reason - CI did not fail even though there's:
`1 file would be reformatted.`

are we missing some `set -e` somewhere...?

---


<!-- Sent from Reviewable.io -->

karczex reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: all files reviewed, 12 unresolved discussions (waiting on @karczex and @lukaszstolarczuk)

___
*[run_benchmark.py, line 45 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdn0Fl6JEyaI_uhmim:-MWis8B3AuPyQhMv3ul4:b-f816nm) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/run_benchmark.py#L45)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

do we need this?
</blockquote></details>

yes. This method is whole point of this class :) 
Probably may remove __str__(), but I like copyable output in log.
___
*[tests/test.py, line 14 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdp39s73uEohddaT9O:-MWiqpS76vy7v-XRn41q:b-666yrr) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L14)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

move imports to top...?
</blockquote></details>

it is "on top" in some way :) 
I don't want to make (at least now) run_benchmark the package. So, to import it I need to add project_path to PATH 
___
*[tests/test.py, line 68 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdopBw-q3ELsDQ1z-B:-MWirTscAkFZ8E-nurSQ:b-z4tsir) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L68)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

maybe add here schema validation, as well?
</blockquote></details>

schema validation is part of run_bench. Instead of this, I may add separate test with wrong input, and check if exception was thrown. 


<!-- Sent from Reviewable.io -->

KFilipek reviewed (commented):
> Reviewed 3 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: all files reviewed, 13 unresolved discussions (waiting on @karczex and @lukaszstolarczuk)

___
*[bench_scenarios/generate_dram_scope.py, line 29 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MXCKfhA6KXbnoAv0hJS:-MXCKfhA6KXbnoAv0hJT:b-rdw2ux) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L29)):*
> ```Python
> value_size = [8, 128, 1024]
> number_of_elements = 1000000
> ```

remove extra lines if not necessary


<!-- Sent from Reviewable.io -->

igchor reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: all files reviewed, 14 unresolved discussions (waiting on @karczex and @lukaszstolarczuk)

___
*[bench_scenarios/generate_dram_scope.py, line 28 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnD564yG72dsqkUj2:-MYJOqam-avCnpFh1ywP:b7fb1eg) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L28)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

only one Mill. ? perhaps more...?
</blockquote></details>

Yeah, let's make it 10M like in our benchmarks (for all scopes).


<!-- Sent from Reviewable.io -->

Commit:
Add emon handler to benchmark runner

Commit:
Move uploading data to mongodb to separate scripts

Commit:
Move validation against schema from scenario generator to tests

Commit:
Update benchmarking scenarios

including running DRAM-only engines. It implies to not require 'db' param anymore.

igchor reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: 1 of 11 files reviewed, 16 unresolved discussions (waiting on @igchor, @karczex, @KFilipek, and @lukaszstolarczuk)

___
*[run_benchmark.py, line 243 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYsVr-b5cjXZy8TRC2S:-MYsVr-c61pcOLeik3AG:b-o6bwj5) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L243)):*
> ```Python
> 
>     def get_results(self):
>         def preprocess(d):
> ```

maybe some comment?
___
*[bench_scenarios/generate_memkind_based_scope.py, line 24 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYsX7NY2QsF7Io38PsU:-MYsX7NY2QsF7Io38PsV:bwemvvw) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/bench_scenarios/generate_memkind_based_scope.py#L24)):*
> ```Python
> key_size = [8]
> value_size = [8, 128, 256, 512, 1024]
> number_of_elements = 100000000
> ```

100M? maybe make it 10M like for dram/obj?
___
*[utils/mongodb/upload_to_mongo.sh, line 8 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYsXbr4AGyofneGd50h:-MYsXbr4AGyofneGd50i:b8xetow) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.sh#L8)):*
> ```Shell
> json_files=$(find results -iname "result.json")
> for f in ${json_files}; do
> 	cat ${f} | sed -e 's/"emon": 1/"emon": "1"/g' | sed -e 's/"emon": 0/"emon": "0"/g' | sed -e 's/"pmemkv_bench": {/"params": {/g' > ${f}_parsed.json
> ```

this sed is not longer necessary I guess


<!-- Sent from Reviewable.io -->

karczex reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: 1 of 11 files reviewed, 16 unresolved discussions (waiting on @igchor, @karczex, @KFilipek, and @lukaszstolarczuk)

___
*[run_benchmark.py, line 27 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdn1RH1q3t9Tcfvs50:-MYsxaXg4kZe1SFyb9Hj:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/run_benchmark.py#L27)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

maybe just a short description, it's not obvious what's this for (I guess)
</blockquote></details>

Done.
___
*[run_benchmark.py, line 45 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdn0Fl6JEyaI_uhmim:-MYsxb3M6ygxJO4cs1pe:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/run_benchmark.py#L45)):*
<details><summary><i>Previously, karczex (Paweł Karczewski) wrote…</i></summary><blockquote>

yes. This method is whole point of this class :) 
Probably may remove __str__(), but I like copyable output in log.
</blockquote></details>

Done.
___
*[run_benchmark.py, line 161 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdkVES3BivFmDTZFoM:-MXli0ilDTOo0A592cPJ:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/run_benchmark.py#L161)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

perhaps rename to `numactl_params`
</blockquote></details>

Done.
___
*[bench_scenarios/generate_dram_scope.py, line 12 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdjT9V96dJQdWjFy8H:-MYjXg4n9UjtBixD9AW1:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L12)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

\- jsonschema
</blockquote></details>

Done.
___
*[bench_scenarios/generate_dram_scope.py, line 28 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnD564yG72dsqkUj2:-MYjXnbV12CF9Xm0_Kl0:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L28)):*
<details><summary><i>Previously, igchor (Igor Chorążewicz) wrote…</i></summary><blockquote>

Yeah, let's make it 10M like in our benchmarks (for all scopes).
</blockquote></details>

Done.
___
*[bench_scenarios/generate_dram_scope.py, line 29 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MXCKfhA6KXbnoAv0hJS:-MYjY8Ct4yrFFu5r-0iX:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L29)):*
<details><summary><i>Previously, KFilipek (Krzysztof Filipek) wrote…</i></summary><blockquote>

remove extra lines if not necessary
</blockquote></details>

Done.
___
*[bench_scenarios/generate_dram_scope.py, line 51 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnQgQ8F8Px9ULwo9q:-MYjYXut8Hd3cqzT92oU:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_dram_scope.py#L51)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

missing key_size
</blockquote></details>

Done.
___
*[bench_scenarios/generate_memkind_based_scope.py, line 30 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdnXyk6jnSmxDVE6lv:-MYsxx2y-Y6UGcvTJlEE:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_memkind_based_scope.py#L30)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

empty line, pls remove
</blockquote></details>

Done.
___
*[bench_scenarios/generate_memkind_based_scope.py, line 67 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdno2r0UM4KrgjQ2nL:-MYsxzfD1nMyOgVVWBNA:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_memkind_based_scope.py#L67)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

make it a const on top, pls
</blockquote></details>

Done.
___
*[bench_scenarios/generate_obj_based_scope.py, line 76 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdo8lf0bx4YAcEPF1Q:-MYsy3229TanslyVgbMG:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/bench_scenarios/generate_obj_based_scope.py#L76)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.
___
*[tests/test.py, line 14 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdp39s73uEohddaT9O:-MYsy4e999S-IrcBj4nG:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L14)):*
<details><summary><i>Previously, karczex (Paweł Karczewski) wrote…</i></summary><blockquote>

it is "on top" in some way :) 
I don't want to make (at least now) run_benchmark the package. So, to import it I need to add project_path to PATH 
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

lukaszstolarczuk reviewed (commented):
> Reviewed 6 of 10 files at r2.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: 7 of 11 files reviewed, 21 unresolved discussions (waiting on @igchor, @karczex, @KFilipek, and @lukaszstolarczuk)

*[a discussion](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdq49L2jbYl-aQK219:-MYtJw6417KsRjG8EJn3:brvupug) (no related file):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

hmm and for some reason - CI did not fail even though there's:
`1 file would be reformatted.`

are we missing some `set -e` somewhere...?
</blockquote></details>

did you look into this, perhaps?

at least, pls apply pyformat for your files

---
___
*[run_benchmark.py, line 66 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtLwkn1M6EMCRzvss1:-MYtLwko0seG2Z-bFGwj:b-ojhv13) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L66)):*
> ```Python
>         logger.info("Stop emon")
>         subprocess.run("emon -stop", shell=True)
>         self._emon_process.wait(timeout=timeout)
> ```

can't you use timeout as a `run` param?
___
*[run_benchmark.py, line 75 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtMfM7E_bBWJ4N_dyL:-MYtMfM7E_bBWJ4N_dyM:b1vqalh) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L75)):*
> ```Python
>         return None
> 
>     def get_emon_v(self):
> ```

is it used anywhere?
___
*[run_benchmark.py, line 257 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtOQN_3N5QxhfHhdIA:-MYtOQN_3N5QxhfHhdIB:b-s9a3ez) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L257)):*
> ```Python
> 
> def print_results(results_dict):
>     print(json.dumps(results_dict, indent=4, sort_keys=True))
> ```

pls add some header/description - what's being printed
___
*[run_benchmark.py, line 260 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtNKcG5APPo5TKd24x:-MYtNKcG5APPo5TKd24y:b-my3e4) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L260)):*
> ```Python
> 
> 
> def save_results(results_dict, emon_dat=None):
> ```

I know we produce "emon.dat", but perhaps rename `emon_dat` to `emon_output` (?)
___
*[run_benchmark.py, line 327 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtN1Kb0ghRyf-0lB8q:-MYtN1Kb0ghRyf-0lB8r:bfsrvyl) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L327)):*
> ```Python
> | |                        |          |       +-----------------------+
> | +------------------------+          | 
> +-------------------------------------+ 
> ```

whitespaces here and above (at the end of lines)
___
*[bench_scenarios/generate_memkind_based_scope.py, line 53 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtJlXj3_aSBwyb3tre:-MYtJlXk9sJLdW75xBNG:b-r103hs) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/bench_scenarios/generate_memkind_based_scope.py#L53)):*
> ```Python
>     benchmarks.extend(concurrent_engines())
>     benchmarks_configuration = []
>     db_path = os.getenv("PMEMKV_BENCH_DB_PATH", "/mnt/pmem0")
> ```

all this new env vars will have to be described somewhere (in the next PR, as discussed)
___
*[bench_scenarios/generate_obj_based_scope.py, line 59 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtJOkY1s2wM6HtRm7s:-MYtJOkY1s2wM6HtRm7t:bs6ck06) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/bench_scenarios/generate_obj_based_scope.py#L59)):*
> ```Python
> 
> def generate():
>     benchmarks = []
> ```

this variable is named the same as the global list of `benchmarks` - perhaps rename to `scenarios`
___
*[tests/test.py, line 14 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdp39s73uEohddaT9O:-MYtI9LGAHc56afljD3o:bg93un3) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L14)):*
<details><summary><i>Previously, karczex (Paweł Karczewski) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

och, make sense
___
*[tests/test.py, line 34 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdosIIBdKt8HLk6Xxx:-MYtIDMg0rYVNF94nViv:bb7dsoj) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L34)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

perhaps add also DRAM engine...?
</blockquote></details>

boop
___
*[tests/test.py, line 68 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdopBw-q3ELsDQ1z-B:-MYtImovEMgXWhQSC8O8:bvyse5v) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L68)):*
<details><summary><i>Previously, karczex (Paweł Karczewski) wrote…</i></summary><blockquote>

schema validation is part of run_bench. Instead of this, I may add separate test with wrong input, and check if exception was thrown. 
</blockquote></details>

I agree, you could :)
___
*[tests/test.py, line 144 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtIX6qEB_6W9JVb5QT:-MYtIX6qEB_6W9JVb5QU:bl6e0au) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/tests/test.py#L144)):*
> ```Python
>         (
>             "vcmap",
>             os.path.dirname(os.getenv("KV_BENCH_TEST_PATH", "/dev/shm/pmemkv")),
> ```

now, while I'm looking at this - shouldn't the default paths for vcmap/vsmap be a directory, not a file... ;) could you pls fix this, btw
___
*[utils/emon/postprocess_emon.sh, line 7 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtHEzr621fGjpMkI0_:-MYtHEzr621fGjpMkI0a:bthazv4) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/emon/postprocess_emon.sh#L7)):*
> ```Shell
> 
> edp_config_path=$(dirname ${BASH_SOURCE[0]})/edp_config.txt
> pushd ${PMEMKV_BENCH_RESULTS_PATH} 
> ```

whitespace at the end of this line
___
*[utils/emon/postprocess_emon.sh, line 11 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtHIA60759CnNbsRYr:-MYtHIA60759CnNbsRYs:bn83b6w) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/emon/postprocess_emon.sh#L11)):*
> ```Shell
> for f in ${emon_files}; do
> 	pushd $(dirname ${f})
> 	emon -process-edp ${edp_config_path} 		
> ```

. (tabs here)
___
*[utils/mongodb/upload_to_mongo.py, line 23 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtOuse8K-jbRk_yYaV:-MYtOuse8K-jbRk_yYaW:b3bl87k) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.py#L23)):*
> ```Python
> 
> 
> def preprocess(d):
> ```

pls add a comment, what's this for
___
*[utils/mongodb/upload_to_mongo.py, line 27 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtOsHY2b69K3Bvf91-:-MYtOsHY2b69K3Bvf910:b-nwkdrl) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.py#L27)):*
> ```Python
>         for k, v in d.items():
>             return {
>                 ("metric" if "name" in k else k.replace(".", "_")): preprocess(v)
> ```

I've seen part of this code somewhere... ;)
___
*[utils/mongodb/upload_to_mongo.py, line 49 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtGOKb25d1RzubTWpw:-MYtGOKb25d1RzubTWpx:bthazv4) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.py#L49)):*
> ```Python
> if __name__ == "__main__":
>     help_msg = """
> Custom uploader to mongodb. 
> ```

whitespace at the end of this line
___
*[utils/mongodb/upload_to_mongo.py, line 83 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtKcjG7hNNLG2Ay4nX:-MYtKcjG7hNNLG2Ay4nY:b-rpa1ej) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.py#L83)):*
> ```Python
>             data[os.path.basename(csv_path)] = csv_load(csv_file)
>     if args.v:
>         print(json.dumps(data, indent=4, sort_keys=True))
> ```

perhaps add some "header" to this print, to know what is being printed ;)
___
*[utils/mongodb/upload_to_mongo.sh, line 9 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtFoYv-3bUEOSPNaQ9:-MYtFoYw11UDnOj2FxtX:b-ym554e) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.sh#L9)):*
> ```Shell
> for f in ${json_files}; do
> 	cat ${f} | sed -e 's/"emon": 1/"emon": "1"/g' | sed -e 's/"emon": 0/"emon": "0"/g' | sed -e 's/"pmemkv_bench": {/"params": {/g' > ${f}_parsed.json
> 	/root/workspace/pmemkv-bench/upload_to_mongo.py ${f}_parsed.json
> ```

should we have this hardcoded path here? ;)


<!-- Sent from Reviewable.io -->

lukaszstolarczuk reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: 7 of 11 files reviewed, 20 unresolved discussions (waiting on @igchor, @karczex, @KFilipek, and @lukaszstolarczuk)

*[a discussion](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdq49L2jbYl-aQK219:-MZRapS_DBM3FJpQmd3G:b-2jke4s) (no related file):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

did you look into this, perhaps?

at least, pls apply pyformat for your files
</blockquote></details>

this is fixed in #76 

---


<!-- Sent from Reviewable.io -->

KFilipek reviewed (commented):
> Reviewed 1 of 6 files at r1, 7 of 10 files at r2.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: 10 of 11 files reviewed, 19 unresolved discussions (waiting on @igchor and @karczex)



<!-- Sent from Reviewable.io -->

Commit:
Add unit test for json validation

Commit:
Extend help message and logged information in run_benchmark.py

karczex reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: 3 of 12 files reviewed, 19 unresolved discussions (waiting on @igchor, @karczex, @KFilipek, and @lukaszstolarczuk)

___
*[run_benchmark.py, line 75 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtMfM7E_bBWJ4N_dyL:-MfpvyYt9nd7yjQ3HdnQ:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L75)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

is it used anywhere?
</blockquote></details>

Done.
___
*[run_benchmark.py, line 243 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYsVr-b5cjXZy8TRC2S:-MZCKliTDlxYtwdObbK7:brpmqc3) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L243)):*
<details><summary><i>Previously, igchor (Igor Chorążewicz) wrote…</i></summary><blockquote>

maybe some comment?
</blockquote></details>

removed.
___
*[run_benchmark.py, line 260 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtNKcG5APPo5TKd24x:-MfpwZsT0E71bG6KV_vH:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L260)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

I know we produce "emon.dat", but perhaps rename `emon_dat` to `emon_output` (?)
</blockquote></details>

Done.
___
*[run_benchmark.py, line 327 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtN1Kb0ghRyf-0lB8q:-MfpwavA0LUHmu5-dz4f:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/run_benchmark.py#L327)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

whitespaces here and above (at the end of lines)
</blockquote></details>

Done.
___
*[bench_scenarios/generate_memkind_based_scope.py, line 24 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYsX7NY2QsF7Io38PsU:-MfpzBML-bQONVFVqI28:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/bench_scenarios/generate_memkind_based_scope.py#L24)):*
<details><summary><i>Previously, igchor (Igor Chorążewicz) wrote…</i></summary><blockquote>

100M? maybe make it 10M like for dram/obj?
</blockquote></details>

Done.
___
*[bench_scenarios/generate_obj_based_scope.py, line 59 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtJOkY1s2wM6HtRm7s:-Mfq6X-s4B0TZd42LWmn:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/bench_scenarios/generate_obj_based_scope.py#L59)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

this variable is named the same as the global list of `benchmarks` - perhaps rename to `scenarios`
</blockquote></details>

Done.
___
*[tests/test.py, line 34 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdosIIBdKt8HLk6Xxx:-Mfq6doCBFcnaAbHHTbU:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L34)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

boop
</blockquote></details>

Done.
___
*[tests/test.py, line 68 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdopBw-q3ELsDQ1z-B:-Mfq6gv312wvMIk3YZMf:bshm7ek) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L68)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

I agree, you could :)
</blockquote></details>

Unit tests added. 
___
*[tests/test.py, line 144 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtIX6qEB_6W9JVb5QT:-Mfq76C2Ems5L02WmGyF:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/tests/test.py#L144)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

now, while I'm looking at this - shouldn't the default paths for vcmap/vsmap be a directory, not a file... ;) could you pls fix this, btw
</blockquote></details>

Done.
___
*[utils/emon/postprocess_emon.sh, line 7 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtHEzr621fGjpMkI0_:-Mfq7EcwCh8S3696yDNg:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/emon/postprocess_emon.sh#L7)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

whitespace at the end of this line
</blockquote></details>

Done.
___
*[utils/emon/postprocess_emon.sh, line 11 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtHIA60759CnNbsRYr:-Mfq7G9x0GNBOHkgU5Sl:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/emon/postprocess_emon.sh#L11)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

. (tabs here)
</blockquote></details>

Done.
___
*[utils/mongodb/upload_to_mongo.py, line 23 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtOuse8K-jbRk_yYaV:-Mfq88KI9N9gzBRinHJI:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.py#L23)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

pls add a comment, what's this for
</blockquote></details>

Done.
___
*[utils/mongodb/upload_to_mongo.py, line 49 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtGOKb25d1RzubTWpw:-Mfq7WcX8cx9Sv9RdSV5:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.py#L49)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

whitespace at the end of this line
</blockquote></details>

Done.
___
*[utils/mongodb/upload_to_mongo.py, line 83 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtKcjG7hNNLG2Ay4nX:-Mfq8pha7hhjjnMSBgof:bourx6b) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.py#L83)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

perhaps add some "header" to this print, to know what is being printed ;)
</blockquote></details>

This program can upload any json or csv to the database. It's not possible to predict what would be printed.
___
*[utils/mongodb/upload_to_mongo.sh, line 8 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYsXbr4AGyofneGd50h:-MfqBwkn3VmuFIQ1iVlu:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.sh#L8)):*
<details><summary><i>Previously, igchor (Igor Chorążewicz) wrote…</i></summary><blockquote>

this sed is not longer necessary I guess
</blockquote></details>

Done.
___
*[utils/mongodb/upload_to_mongo.sh, line 9 at r2](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MYtFoYv-3bUEOSPNaQ9:-MfqBzvOCes7YmZSKw3I:b-896fix) ([raw file](https://github.com/pmem/pmemkv-bench/blob/fda0ce3c0fcc35622a006177b3f1381af1d5a01b/utils/mongodb/upload_to_mongo.sh#L9)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

should we have this hardcoded path here? ;)
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

igchor reviewed (commented):
> Reviewed 2 of 9 files at r3.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: 5 of 12 files reviewed, 16 unresolved discussions (waiting on @karczex, @KFilipek, and @lukaszstolarczuk)



<!-- Sent from Reviewable.io -->

lukaszstolarczuk committed:
CI: update build script executed in docker

- enable more verbose logs,
- export test path variable into environment,
- enable fast tests' failing.

lukaszstolarczuk committed:
CI: execute tests on /tmp not on /dev/shm by default

and use different default file name, since 'pmemkv' may accidentally exists there.

lukaszstolarczuk reviewed (commented):
I've rebased the changes with Janek's https://github.com/pmem/pmemkv-bench/pull/73 \+ changed the actual testing directories to ones specified in tests, not the one specified in CI.

> Reviewed 1 of 9 files at r3, 1 of 12 files at r4, 1 of 2 files at r5, 12 of 12 files at r6, 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-:-MgFzfhwDvvNlH-OXSBc:b74lkc5)* status: all files reviewed, 1 unresolved discussion (waiting on @karczex)

___
*[tests/test.py, line 68 at r1](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MWdopBw-q3ELsDQ1z-B:-MgAsypc9SavQRSpDuw7:b2nid0f) ([raw file](https://github.com/pmem/pmemkv-bench/blob/a2cc24399dff1d472497deabab3dd31a62099833/tests/test.py#L68)):*
<details><summary><i>Previously, karczex (Paweł Karczewski) wrote…</i></summary><blockquote>

Unit tests added. 
</blockquote></details>

thx
___
*[tests/test.py, line 140 at r7](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MgFx1w6CcsfAMqxCjH-:-MgFx1w6CcsfAMqxCjH0:by61ayt) ([raw file](https://github.com/pmem/pmemkv-bench/blob/5669192b48a20115acf078ffc7a125ea377311af/tests/test.py#L140)):*
> ```Python
>         (
>             "cmap",
>             os.getenv("KV_BENCH_TEST_PATH", "/tmp/pmemkv"),
> ```

just FYI, I couldn't make docker's `/dev/shm` bigger (in an easy way; GHA currently does not support run/build args for Docker in their modules - I'd have to build dockers with custom scripts)
___
*[tests/test.py, line 183 at r7](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-MgFxZOO3q9kDLsyUnw6:-MgFxZOO3q9kDLsyUnw7:bglqjco) ([raw file](https://github.com/pmem/pmemkv-bench/blob/5669192b48a20115acf078ffc7a125ea377311af/tests/test.py#L183)):*
> ```Python
>                 "--threads": "2",
>             },
>             "cleanup": 1,
> ```

For some reason, I kept getting the error, without setting up cleanup in all tests.
```
Running readrandom\nCannot start engine 'cmap' for path '/tmp/pmemkv' with 1 GB capacity.\nError '[pmemkv_open] Failed creating pool: file /tmp/pmemkv already exists'\n
```

Even though we have `create_if_missing` flag set up (https://github.com/pmem/pmemkv-bench/blob/master/bench/db_bench.cc#L765)

Any ideas why..?


<!-- Sent from Reviewable.io -->

igchor reviewed (commented):
LGTM

> Reviewed 1 of 12 files at r4, 5 of 12 files at r6.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-:-MgLAfFh5yUBiAg1tC5Y:bb6ywmx)* status: all files reviewed, 1 unresolved discussion (waiting on @karczex)



<!-- Sent from Reviewable.io -->

KFilipek reviewed (approved):
:LGTM:
> Reviewed 1 of 12 files at r4, 1 of 2 files at r5, 11 of 12 files at r6, 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: all files reviewed, 1 unresolved discussion (waiting on @karczex)



<!-- Sent from Reviewable.io -->

lukaszstolarczuk committed:
Update Makefile

'make bench' should only clean compiled bench file, not reset the test path.

lukaszstolarczuk committed:
Update default bench's test path

lukaszstolarczuk committed:
Properly use KV_BENCH_TEST_PATH in tests

and set this env in run_benchmark.py

lukaszstolarczuk reviewed (commented):
I'm not sure how these tests worked before (on master):
on one hand, we accidentally set up the same test path as the default one (for `make reset`)
on the other hand, we introduced a new cleanup flag and it wasn't used (?)

The tests passed before, so I guess I still don't understand something :wink: 


**Anyway**, right now, I believe it works as expected - we only cleanup the pool file if `cleanup` flag is set, so it is done as described in README - cleaning up the pool is the user's responsibility, not makefile's.

> Reviewed 3 of 5 files at r9, 10 of 12 files at r10, 4 of 4 files at r11.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68#-:-MgRxyb27zehxwSbctox:b77akn3)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @karczex)



<!-- Sent from Reviewable.io -->

KFilipek reviewed (approved):
> Reviewed 3 of 5 files at r9, 10 of 12 files at r10, 4 of 4 files at r11.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv-bench/68)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @karczex)



<!-- Sent from Reviewable.io -->
